### PR TITLE
feat: new 404 page

### LINKED ISSUE #1259: Fehlerscreens

### PR DESCRIPTION
**Pull request by Schwehn42**
## Description

This PR implements the new 404 error page.  
 Resolves #1259

## Changelog

* Change layout and CSS as per design
* Add required assets
* Make design responsive to size and theme

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The light- and dark-theme are both supported and tested
- [x] The design was implemented and is responsive for all devices and screen sizes
- [x] The application was tested in the most commonly used browsers (e.g. Chrome, Firefox, Safari)

## (Optional) Visual Changes
Bildschirm­foto 2023-04-24 um 13 21 03
Bildschirm­foto 2023-04-24 um 13 25 38

### CODE REVIEW

File: src/routes/NotFound/NotFound.scss [hunk 1]
```diff
@@ -1,26 +1,185 @@
...
+
+.not-found__description {
+  margin-top: $margin--default;
+
+  font-weight: 400;
+  font-size: 38px;
+  line-height: 50px;
+}
+
+.not-found__return-button {
```
Comment (Resaki1) [inline src/routes/NotFound/NotFound.scss:70]: I'd personally center the button on the screen for mobile devices. What do you think?

**Review by Resaki1 — CHANGES_REQUESTED** [review]
On some mobile screens, especially smaller ones (iPhone SE), the content is cut off at the top and there is no space between the return button and the bottom of the screen. This is especially noticeable when holding the phone sideways, where the button can't even be reached

File: src/routes/NotFound/NotFound.scss [hunk 2]
```diff
@@ -1,26 +1,185 @@
...
+@import "src/constants/style";
 
-$logo-height: 60px;
-$text-size--medium: 24px;
+.not-found__root {
+  height: 100%;
+  width: 100%;
+  padding: 20px 0 20px 20px;
+  overflow-x: hidden;
+  background: linear-gradient(320deg, rgb(107, 156, 255, 30%), rgb(255, 255, 255, 30%));
```
[the remark this answers is not in the document]
Comment (Schwehn42) [reply src/routes/NotFound/NotFound.scss:8]: ah nice, didn't know about that! that's quite helpful :)

File: src/routes/NotFound/NotFound.scss [hunk 3]
```diff
@@ -1,26 +1,185 @@
...
+  font-size: 38px;
+  line-height: 50px;
+}
+
+.not-found__return-button {
+  margin-top: 90px;
+
+  padding: 20px 40px;
   color: $color-white;
+  background: #ff0069;
```
[the remark this answers is not in the document]
Comment (Schwehn42) [reply src/routes/NotFound/NotFound.scss:75]: planning pink (`#e20360`) is slightly darker than `#ff0069` (which the design uses), and I also think the latter looks better on the button 😅

File: src/routes/NotFound/NotFound.scss [hunk 4]
```diff
@@ -1,26 +1,185 @@
...
+
+  .not-found__logo-stan--mobile {
+    display: none;
+  }
+}
+
+@media screen and (min-width: 2048px) {
+  .not-found__background-form {
+    top: -100px;
+    left: 0;
```
[the remark this answers is not in the document]
Comment (Schwehn42) [reply src/routes/NotFound/NotFound.scss:158]: actually, this is on purpose, [the design](https://www.figma.com/file/8m2V6OJTImuOdQoxruWurP/Scrumlr?node-id=616-18178&t=uZvcHuiL2rWuV80x-0) specifies it that way.  
However, I also agree that this looks kinda odd 😅 I also took the liberty of slightly customizing the design (i.e. always putting the scrumlr on the top left), so might as well change the background-form so it looks right

File: src/routes/NotFound/NotFound.scss [hunk 5]
```diff
@@ -1,26 +1,185 @@
...
+
+  font-weight: 400;
+  font-size: 38px;
+  line-height: 50px;
+}
+
+.not-found__return-button {
+  margin-top: 90px;
+
+  padding: 20px 40px;
```
[the remark this answers is not in the document]
Comment (Schwehn42) [reply src/routes/NotFound/NotFound.scss:73]: hmm, I tried using the different preexisiting paddings, but the combinations make the button look either too slim or too bulbuos 👀

File: src/routes/NotFound/NotFound.tsx [hunk 6]
```diff
@@ -1,10 +1,59 @@
...
+        <BackgroundFreeFormLight className="not-found__background-form not-found__background-form--light" />
+        <BackgroundFreeFormDark className="not-found__background-form not-found__background-form--dark" />
+        <BackgroundDetails className="not-found__background-details" />
+      </div>
+      <header className="not-found__header">
+        <div className="scrumlr-logo">
+          <ScrumlrLogoWebLight className="not-found__scrumlr-logo not-found__scrumlr-logo--web not-found__scrumlr-logo--light" />
+          <ScrumlrLogoWebDark className="not-found__scrumlr-logo not-found__scrumlr-logo--web not-found__scrumlr-logo--dark" />
+          <ScrumlrLogoMobileLight className="not-found__scrumlr-logo not-found__scrumlr-logo--mobile not-found__scrumlr-logo--light" />
+          <ScrumlrLogoMobileDark className="not-found__scrumlr-logo not-found__scrumlr-logo--mobile not-found__scrumlr-logo--dark" />
```
[the remark this answers is not in the document]
Comment (Schwehn42) [reply src/routes/NotFound/NotFound.tsx:35]: oops forgot about that one ^^ i'll change it

**Comment by Schwehn42** [comment]
@Resaki1 the page should be _mostly_ responsive now. Phones that are really old (like iPhone 4, Nexus 2) and so on still might have issues, but any phone with a reasonable age looks good as far as I can tell

File: src/routes/NotFound/NotFound.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (brandstetterm) [reply src/routes/NotFound/NotFound.scss:72]: I think this should be discussed
Using #ff0069 would be another color in our color palette and also adds inconsistencies with the planning pink color used on our boards

File: src/routes/NotFound/NotFound.scss [hunk 7]
```diff
@@ -1,26 +1,194 @@
...
+
+.not-found__return-button {
+  margin-top: 20px;
+
+  padding: 15px 30px;
   color: $color-white;
+  background: #ff0069;
+
+  font-weight: 700;
+  font-size: $text-size--large;
```
Comment (brandstetterm) [inline src/routes/NotFound/NotFound.scss:75]: In my opinion either the font size or the button in general looks too big
image

Comment (brandstetterm) [reply src/routes/NotFound/NotFound.scss:75]: I think it looks better with `font-size: 16px`

File: src/routes/NotFound/NotFound.scss [hunk 8]
```diff
@@ -1,26 +1,194 @@
...
+  font-weight: 400;
+  font-size: 24px;
+  line-height: 30px;
 }
-[theme="dark"] .not-found__text {
+
+.not-found__return-button {
+  margin-top: 20px;
+
+  padding: 15px 30px;
```
Comment (brandstetterm) [inline src/routes/NotFound/NotFound.scss:70]: Please use our padding constants

File: src/routes/NotFound/NotFound.scss [hunk 9]
```diff
@@ -1,26 +1,194 @@
...
+  margin-top: $margin--small;
+
+  font-weight: 400;
+  font-size: 24px;
+  line-height: 30px;
 }
-[theme="dark"] .not-found__text {
+
+.not-found__return-button {
+  margin-top: 20px;
```
Comment (brandstetterm) [inline src/routes/NotFound/NotFound.scss:68]: Please use our margin constants here
For example `margin-top: $margin--default`

File: src/routes/NotFound/NotFound.scss [hunk 10]
```diff
@@ -1,26 +1,194 @@
...
+.not-found__title {
+  font-weight: 700;
+  font-size: 45px;
+}
+
+.not-found__description {
+  margin-top: $margin--small;
+
+  font-weight: 400;
+  font-size: 24px;
```
Comment (brandstetterm) [inline src/routes/NotFound/NotFound.scss:63]: According to the design the description text size should be approximately twice the size of the button, so I guess we should either change the font size of the description or button

File: src/routes/NotFound/NotFound.scss [hunk 11]
```diff
@@ -1,26 +1,194 @@
...
+}
+
+.not-found__background {
+  position: relative;
+}
+
+.not-found__background-form {
+  position: absolute;
+  top: -440px;
+  left: -960px;
```
Comment (brandstetterm) [inline src/routes/NotFound/NotFound.scss:19]: Can you maybe add a comment that explains those magic numbers?

File: src/routes/NotFound/NotFound.scss
Comment (brandstetterm) [inline src/routes/NotFound/NotFound.scss:1]: I've sat down with @SelinaBuff this afternoon to discuss the font sizes. The result has been that it looks best with: 

`.not-found__description` should have `font-size: 24px` and `line-height: 40px` and a margin of 12px to `.not-found__title`. 

`.not-found__return-button` should have `font-size: 16px` and a margin of 20px to `.not-found__description`

I hope I've remembered everything correctly 😅

Comment (brandstetterm) [reply src/routes/NotFound/NotFound.scss:1]: Should be looking like that:
Screenshot 2023-05-05 at 14 32 22

Comment (Schwehn42) [reply src/routes/NotFound/NotFound.scss:1]: Also, how large is the title supposed to be?

File: src/routes/NotFound/NotFound.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (Schwehn42) [reply src/routes/NotFound/NotFound.scss:72]: I could
1. Use the exisiting color planning pink for the button
2. Use the exisiting color planning pink for the button but modifiy it using SCSS color functions to look kinda like the other color
3. Add the other color to the palette

What do you prefer?

File: src/routes/NotFound/NotFound.scss (hunk 10 above)
Comment (Schwehn42) [reply src/routes/NotFound/NotFound.scss:66]: Alright, I'll use these values then: https://github.com/inovex/scrumlr.io/pull/2899#discussion_r1186456810

File: src/routes/NotFound/NotFound.scss
Comment (brandstetterm) [reply src/routes/NotFound/NotFound.scss:1]: The new font size constants coming with this PR should be used https://github.com/inovex/scrumlr.io/pull/2961

**Comment by Schwehn42** [comment]
@brandstetterm I used the new size constants where it's possible, it should mostly look good now I think

File: src/routes/NotFound/NotFound.scss [hunk 12]
```diff
@@ -1,26 +1,196 @@
...
+.not-found__title {
+  font-weight: 700;
+  font-size: $text--2xl;
+}
+
+.not-found__description {
+  margin-top: 12px;
+
+  font-weight: 400;
+  font-size: $text-size--large;
```
Comment (brandstetterm) [inline src/routes/NotFound/NotFound.scss:65]: Please use one of the new constants

File: src/routes/NotFound/NotFound.scss [hunk 13]
```diff
@@ -1,26 +1,196 @@
...
+}
+
+// default look is for very small screens and works its way up to match larger screen sizes (could theoretically do it the other way around)
+@media screen and (min-height: 600px) {
+  .not-found__description {
+    margin-top: $margin--default;
+  }
+
+  .not-found__return-button {
+    margin-top: $margin--large + $margin--small; // 40px
```
Comment (brandstetterm) [inline src/routes/NotFound/NotFound.scss:105]: I think it looks better if you always use `padding: $padding--default $padding--large;`

**Pull request merged**